contracts/exchange/test/transactions_unit_tests.ts

```diff
@@ -0,0 +1,707 @@
+                [randomSignature()],
+                { from: accounts[2] },
+            );
+            return expect(tx).to.revertWith(outerExpectedError);
+        });
+        it('should allow recursion as long as currentContextAddress is not set', async () => {
+            const innerTransaction1 = await generateZeroExTransactionAsync({ signerAddress: accounts[0] });
+            const innerTransaction2 = await generateZeroExTransactionAsync({ signerAddress: accounts[1] });
+            // From this point on, all transactions and calls will have the same sender, which does not change currentContextAddress when called
+            const innerBatchExecuteTransaction = await generateZeroExTransactionAsync({
+                signerAddress: accounts[2],
+                callData: transactionsContract.batchExecuteTransactions.getABIEncodedTransactionData(
+                    [innerTransaction1, innerTransaction2],
+                    [randomSignature(), randomSignature()],
+                ),
+            });
+            const outerExecuteTransaction = await generateZeroExTransactionAsync({
+                signerAddress: accounts[2],
+                callData: transactionsContract.executeTransaction.getABIEncodedTransactionData(
+                    innerBatchExecuteTransaction,
+                    randomSignature(),
+                ),
+            });
+            return expect(
+                transactionsContract.batchExecuteTransactions.awaitTransactionSuccessAsync(
+                    [outerExecuteTransaction],
+                    [randomSignature()],
+                    { from: accounts[2] },
+                ),
+            ).to.be.fulfilled('');
+        });
+    });
+
+    describe('executeTransaction', () => {
+        function getExecuteTransactionCallData(transaction: ZeroExTransaction, signature: string): string {
+            return (transactionsContract as any).executeTransaction.getABIEncodedTransactionData(
+                transaction,
+                signature,
+            );
+        }
+        it('should revert if the current time is past the expiration time', async () => {
+            const transaction = await generateZeroExTransactionAsync({
+                expirationTimeSeconds: constants.ZERO_AMOUNT,
+            });
+            const transactionHash = transactionHashUtils.getTransactionHashHex(transaction);
+            const expectedError = new ExchangeRevertErrors.TransactionError(
+                ExchangeRevertErrors.TransactionErrorCode.Expired,
+                transactionHash,
+            );
+            const tx = transactionsContract.executeTransaction.awaitTransactionSuccessAsync(
+                transaction,
+                randomSignature(),
+            );
+            return expect(tx).to.revertWith(expectedError);
+        });
+        it('should revert if the transaction is submitted with a gasPrice that does not equal the required gasPrice', async () => {
```

Comment:
Since you already test this behavior in `assertExecutableTransaction`'s test, I'd be fine if we want to remove this. Any reason why we shouldn't?

Reply:
I feel like most of the other tests are actually covered by the `assertExecutableTransaction` tests. We have extra tests for expired transactions, valid/invalid signatures, already executed transactions, etc. If we remove this, it probably makes sense to remove the other redundant tests as well.

Reply:
I guess out of these options, it might make sense to just keep them to prevent regressions. Longer term, I do think we should try to reduce test duplication, but it's easy to understand which are duplicates in this file.